src/go/rpk/pkg/cli/container/start.go

```diff
@@ -520,10 +600,120 @@ func getContainerErr(state *common.NodeState, c common.Client) (string, error) {
 	// demux this stream we need to use stdcopy.StdCopy. See:
 	// https://github.com/moby/moby/issues/32794#issuecomment-297151440
 	bErr := new(bytes.Buffer)
-	_, err = stdcopy.StdCopy(bErr, bErr, reader)
+	_, err = stdcopy.StdCopy(nil, bErr, reader)
 	if err != nil {
 		return "", fmt.Errorf("unable to read docker logs: %v", err)
 	}
 
 	return bErr.String(), nil
 }
+
+// parsePort parses a single port array.
+//   - If the port array is empty, we return a default port array.
+//   - If the port is 'any' we get a random port using vnet.GetFreePortPool.
+//   - If the provided port array < number of nodes, we fill the port array with succeeding ports.
+func parsePort(ports []string, nNodes, defPort int) ([]uint, error) {
+	var ret []uint
+	// If no port is defined, we use the default as seed node port.
+	if len(ports) == 0 {
+		ret = append(ret, uint(defPort))
+	}
+	for _, p := range ports {
+		// If port == 'any', we assign a random port.
+		if p == "any" {
+			if len(ports) > 1 {
+				return nil, errors.New("cannot specify 'any' with additional ports")
+			}
+			port, err := vnet.GetFreePortPool(nNodes)
+			if err != nil {
+				return nil, fmt.Errorf("unable to assign random ports: %v", err)
+			}
+			return port, nil
+		}
```

Comment:
nit: this check should probably come above the for loop

Reply:
But `ports` is a string slice. And any of the ports in []port can contain a string `any`